lib/screens/v3_api.ex

```diff
@@ -59,6 +66,15 @@ defmodule Screens.V3Api do
     end
   end
 
+  defp update_response_cache(url, response, headers) do
+    date =
+      headers
+      |> Enum.into(%{})
+      |> Map.get("last-modified")
+
+    ScreenApiResponseCache.put(url, {response, date}, ttl: :timer.hours(1))
```

Comment:
I think in theory, because we are always checking with the server whether our data is still fresh, the only reason cache entries should ever expire is because it's possible to reconfigure screens (or take them offline) such that a given URL is no longer accessed at all, in which case the cached response is just taking up space. But per the Nebulex [docs](https://hexdocs.pm/nebulex/Nebulex.Adapters.Local.html), `ttl` is not a mechanism for evicting unused data, as it's only evaluated when the data is retrieved, which in this scenario it never would be:

> The `:ttl` is evaluated on-demand when a key is retrieved, and at that moment if it's expired, then remove it from the cache, hence, it can not be used as eviction method, it is more for keep the integrity and consistency in the cache.

So I don't think we need to set a `ttl` here; the garbage collection should take care of cleaning up responses that aren't being actively used, and the If-Modified-Since checking should ensure we never use a stale response.

What we might want to do is set a maximum `allocated_memory` just to make sure we don't hit the instance's limit — the docs recommend setting either this or `max_size`, and neither is set by default. It appears we have 1GB to work with (in all environments) and currently use about 40% of it, so maybe 250MB for the cache?

Reply:
Interesting, good catch. 250MB sounds good, at least for now. If we see issues with this, we can adjust as needed.